opm/simulators/flow/BlackoilModelEbosNldd.hpp

```diff
@@ -730,20 +738,24 @@ class BlackoilModelEbosNldd {
                             maxres = std::max(maxres, std::fabs(residual[c][ii]));
                         }
                     }
-                    maxres_per_domain[domain.index] = std::make_pair(maxres, domain.index);
-                }
-                // Lexicographical sort by pressure, then index.
-                std::sort(maxres_per_domain.begin(), maxres_per_domain.end());
-                // Reverse since we want high-pressure regions solved first.
-                std::reverse(maxres_per_domain.begin(), maxres_per_domain.end());
-                for (std::size_t ii = 0; ii < domains_.size(); ++ii) {
-                    domain_order[ii] = maxres_per_domain[ii].second;
+                    measure_per_domain[domain.index] = std::make_pair(maxres, domain.index);
                 }
+                break;
             }
-            break;
+            } // end of switch (model_.param().local_domain_ordering_)
+
+
+            // Sort by largest measure, keeping index order if equal.
+            std::stable_sort(measure_per_domain.begin(), measure_per_domain.end(),
+                             [](const auto& m1, const auto& m2){ return m1.first > m2.first; });
```

Comment:
If you're changing the approach anyway, I would probably just sort the `domain_order` directly in all cases other than `Jacobi`.  For instance something along the lines of
```C++
auto domain_order = std::vector<int>(domains_.size());
std::iota(domain_order.begin(), domain_order.end(), 0);
if (Jacobi) {
    return domain_order;
}

auto measure = std::vector<double>(domains_.size());
switch (approach) {
    // fill 'measure' appropriately
}
std::sort(domain_order.begin(), domain_order.end(),
    [&measure](const int i1, const int i2)
{
    return (measure[i1] > measure[i2])
        || ((measure[i1] == measure[i2]) && (i1 < i2));
});
return domain_order;
```
Then we don't have to bother about sorting pairs and we do (slightly) less work and use (slightly) less memory.

Reply:
Follow-up to the previous answer: It's probably better to use
```C++
std::stable_sort(domain_order.begin(), domain_order.end(),
    [&measure](const int i1, const int i2)
{
    return measure[i1] > measure[i2];
});
```
than to use direct equality comparisons for floating-point numbers.